terraform/logging_metric/bucket_counter/main.tf

```diff
@@ -0,0 +1,19 @@
+resource "google_logging_metric" "default" {
+  # Measures the number of times objects are created in a given bucket
+  # Exact trigger (creation, modification, etc.) can be adjusted using the "method" variable.
+  # This implementation was based on https://cloud.google.com/blog/products/storage-data-transfer/guide-to-setting-up-monitoring-for-object-creation-in-cloud-storage
+  # and https://stackoverflow.com/a/37257217
+
+  filter = <<EOT
+resource.type="gcs_bucket"
+resource.labels.bucket_name="${var.bucket_name}"
+protoPayload.resourceName:"${var.prefix}"
```

Comment:
will the semicolon here cause partial matching of the prefix? Or does it only match from the start of the prefix?

Reply:
it's actually a 'has' operator (substring) which isn't necessarily a prefix, thinking of it more I could also use a regular expression to make it behave correctly: https://cloud.google.com/logging/docs/view/advanced-queries#regular-expressions, will make a fix